src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenInParametersTests.cs

```diff
@@ -799,6 +857,57 @@ public static void M1(in int arg1, in int arg2, in int arg3)
         }
 
         [Fact]
+        public void ReadonlyParamAsyncSpillInRoField()
+        {
+            var text = @"
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        public static readonly int F = 5;
+
+        static void Main(string[] args)
+        {
+            Test().Wait();
+        }
+
+        public static async Task Test()
+        {
+            int local = 1;
+            M1(in F, await GetT(2), 3);
+        } 
+
+        public static async Task<T> GetT<T>(T val)
+        {
+            await Task.Yield();
+            MutateReadonlyField();
+            return val;
+        }
+
+        private static unsafe void MutateReadonlyField()
+        {
+            fixed(int* ptr = &F)
+            {
+                *ptr = 42;
+            }
+        }
+
+        public static void M1(in int arg1, in int arg2, in int arg3)
+        {
+            System.Console.WriteLine(arg1 + arg2 + arg3);
+        }
+    }
+
+";
+
+            var comp = CreateCompilationWithMscorlib46(text, new[] { ValueTupleRef, SystemRuntimeFacadeRef }, options: TestOptions.UnsafeReleaseExe);
```

Comment:
Are the references necessary?

Reply:
I just copied from other test. They will not make it a better or worse, so I will let them be. 

---
In reply to: [146965555](https://github.com/dotnet/roslyn/pull/22772#discussion_r146965555) [](ancestors = 146965555)